### PR TITLE
Nicer diagnostics panel - Request

### PR DESCRIPTION
**Pull request by EaredSeal**

### CODE REVIEW

File: src/Kdyby/ElasticSearch/Diagnostics/Panel.php [hunk 1]
```diff
@@ -104,6 +104,11 @@ public function getPanel()
 				return array();
 			}
 
+			if ($object instanceof Elastica\Request) {
+				$j = new Elastica\JSONFormat('  ', "\n"); // indent and linebreak characters
```
Comment (fprochazka) [inline src/Kdyby/ElasticSearch/Diagnostics/Panel.php]: Where is this class from? I can't find it https://github.com/ruflin/Elastica/search?l=php&q=Format&utf8=%E2%9C%93

Comment (EaredSeal) [inline src/Kdyby/ElasticSearch/Diagnostics/Panel.php]: This class Elastica\JSONFormat is in version 2.1.0

Comment (EaredSeal) [inline src/Kdyby/ElasticSearch/Diagnostics/Panel.php]: Now I am ready to upgrade to a request JSON syntax highlighting. Class Elastica \ JSONFormat no longer needed.

File: src/Kdyby/ElasticSearch/Diagnostics/Panel.php [hunk 2]
```diff
@@ -250,4 +255,37 @@ public function register(Kdyby\ElasticSearch\Client $client)
...
+			}
+			elseif(preg_match('/true|false/', $match)) {
+				$cls = 'bool';
+			}
+			elseif(preg_match('/null/', $match)) {
+				$cls = 'null';
+			}
+			return '<span class="tracy-dump-' . $cls . '">' . $match . '</span>';
+		}, $text);
+	}
```
Comment (enumag) [inline src/Kdyby/ElasticSearch/Diagnostics/Panel.php:296]: Add a note where this function is from and fix the coding style.

**Comment by fprochazka** [comment]
I see now what are you doing :) 

Try right clicking on the "request" word (which is in fact a link) before the outputed json and "copy link location". Your clipboard will contain the entire json without formatting.

File: src/Kdyby/ElasticSearch/Diagnostics/Panel.php [hunk 3]
```diff
@@ -250,4 +255,44 @@ public function register(Kdyby\ElasticSearch\Client $client)
...
+
+			$cls = 'number';
+			if (preg_match('/^"/', $match)) {
+				if (preg_match('/:$/', $match)) {
+					$cls = 'key';
+				} else {
+					$cls = 'string';
+				}
+			}
+			elseif (preg_match('/true|false/', $match)) {
```
Comment (enumag) [inline src/Kdyby/ElasticSearch/Diagnostics/Panel.php:287]: don't use newline before elseif

File: src/Kdyby/ElasticSearch/Diagnostics/Panel.php [hunk 4]
```diff
@@ -250,4 +255,44 @@ public function register(Kdyby\ElasticSearch\Client $client)
 		Debugger::getBar()->addPanel($this);
 	}
 
+	/**
+	 * syntax highlighting for rendering Elastica\Request
```
Comment (enumag) [inline src/Kdyby/ElasticSearch/Diagnostics/Panel.php:259]: Did you write this function yourself or did you copy it somewhere?

**Comment by fprochazka** [comment]
I'm thinking instead of doing the highlighting in this way, it may be better to create a simple few lines javascript, that would open new popup with the json unformatted there.

**Pull request merged**